Extension/src/LanguageServer/client.ts

```diff
@@ -1520,6 +1523,9 @@ export class DefaultClient implements Client {
         }
         const workspaceFallbackEncoding: string = workspaceOtherSettings.filesEncoding;
         const lastWorkspaceFallbackEncoding: PersistentState<string> = new PersistentState<string>("CPP.lastWorkspaceFallbackEncoding", "");
+        if (lastWorkspaceFallbackEncoding.Value === "") {
+            lastWorkspaceFallbackEncoding.Value = workspaceFallbackEncoding;
```

Comment:
Same here. Pass the `workspaceFallbackEncoding` as the default value to the constructor.